### PR TITLE
call local_cache 1 time instead of 3 times

### PR DESCRIPTION
**Pull request by grosser**
@rafaelfranca

### CODE REVIEW

**Comment by grosser** [comment]
rebased :)

**Comment by eileencodes** [comment]
Hi @grosser - thanks for the PR! Can you update your commit message to explain a little more indepth how/when it was getting called 3 times. Obviously calling something once is better than 3 times, but commit messages should better explain the what and the why of our changes that way we don't need to come back to github to learn more about a change :smile:

**Comment by grosser** [comment]
better ?

File: activesupport/lib/active_support/cache/strategy/local_cache.rb [hunk 1]
```diff
@@ -102,7 +102,7 @@ def decrement(name, amount = 1, options = nil) # :nodoc:
 
         protected
           def read_entry(key, options) # :nodoc:
-            if local_cache
+            if local_cache = local_cache()
```
Comment (jeremy) [inline activesupport/lib/active_support/cache/strategy/local_cache.rb]: Nice. Clarify by assigning to a local var with a different name than the reader method, even just `if cache = local_cache`

Comment (grosser) [inline activesupport/lib/active_support/cache/strategy/local_cache.rb]: done!

File: activesupport/lib/active_support/cache/strategy/local_cache.rb [hunk 2]
```diff
@@ -102,10 +102,10 @@ def decrement(name, amount = 1, options = nil) # :nodoc:
 
         protected
           def read_entry(key, options) # :nodoc:
-            if local_cache
-              local_cache.fetch(key) do
+            if cache = local_cache()
```
Comment (eileencodes) [inline activesupport/lib/active_support/cache/strategy/local_cache.rb]: You can remove the parentheses here.

Comment (grosser) [inline activesupport/lib/active_support/cache/strategy/local_cache.rb]: done!

**Comment by eileencodes** [comment]
I restarted this job to see if the AJ test passes. Tests after aren't failing and I haven't had time to investigate why they just barf out uninitialized constant on my machine. Can you just check that your change didn't cause the failure in AJ?

**Comment by grosser** [comment]
green now ... so I don't think my change causes the failures

**Comment by eileencodes** [comment]
Ah yes good, I thought I reran it before and it was still red. Didn't think it was related but wanted to double check. 

Thanks!

**Pull request merged**